### PR TITLE
Apply non-controversial parts of ot-style impl

### PR DESCRIPTION
**Pull request by ebraminio**
Things to be used in https://github.com/harfbuzz/harfbuzz/pull/1459

### CODE REVIEW

File: test/shaping/run-tests.py [hunk 1]
```diff
@@ -138,4 +138,5 @@ def cmd(command):
 elif passes:
 	sys.exit (0)
 else:
-	sys.exit (77)
+	# it should raise 77 on skips but that makes logs unusable
+	sys.exit (0)
```
Comment (behdad) [inline test/shaping/run-tests.py]: What??  This is wrong.

File: src/hb-ot-os2-table.hh [hunk 2]
```diff
@@ -93,10 +93,56 @@ struct OS2
 {
   enum { tableTag = HB_OT_TAG_OS2 };
 
+  inline bool has_data () const { return this != &Null (OS2); }
```
Comment (behdad) [inline src/hb-ot-os2-table.hh]: We typically don't like comapring to Null as a way to detect emptyness.  This has legitimate reasons (there might be different null polls for different .so objects, or even files...).  If there's a better way, use it.  Why not the version number, like we do in all other tables?!

File: src/hb-ot-stat-table.hh [hunk 3]
```diff
@@ -223,10 +204,44 @@ struct AxisValue
...
+    return_trace (likely (c->check_struct (this)));
+  }
+
+  protected:
+  Tag		axisTag;	/* A tag identifying the axis of design variation. */
+  NameID	axisNameID;	/* The name ID for entries in the 'name' table that
+				 * provide a display string for this axis. */
+  HBUINT16	axisOrdering;	/* A value that applications can use to determine
+				 * primary sorting of face names, or for ordering
+				 * of descriptors when composing family or face names. */
```
Comment (behdad) [inline src/hb-ot-stat-table.hh]: Typically, members of struct `StatAxisRecord` don't need to have `axis` prefix.  Makes more readable.

File: test/shaping/run-tests.py (hunk 1 above)
Comment (behdad) [reply test/shaping/run-tests.py]: Also, doesn't belong in this PR.

File: src/hb-ot-os2-table.hh (hunk 2 above)
Comment (ebraminio) [reply src/hb-ot-os2-table.hh]: "The version number for the OS/2 table: 0x0000 to 0x0005.", something more reliable please?

Comment (behdad) [reply src/hb-ot-os2-table.hh]: Then we need to add a custom null table for it.   Umm. Keep it as is.  I'll clean up at some point when I'll go over all such checks to Null..

File: src/hb-ot-stat-table.hh [hunk 4]
```diff
@@ -223,10 +204,44 @@ struct AxisValue
...
 {
   enum { tableTag = HB_OT_TAG_STAT };
 
+  inline const AxisValue &get_axis_value (hb_tag_t style) const
+  {
+    const StatAxisRecord &record = (this+designAxesOffset).lsearch (designAxisCount, style);
+    if (!record.has_data () || record.get_ordering () >= axisValueCount)
+      return Null (AxisValue);
+
+    return this+(this+offsetToAxisValueOffsets)[record.get_ordering ()];
```
Comment (ebraminio) [inline src/hb-ot-stat-table.hh]: Plain wrong logic, let me fix...

File: test/shaping/run-tests.py (hunk 1 above)
Comment (ebraminio) [reply test/shaping/run-tests.py]: Gone

File: src/hb-ot-stat-table.hh (hunk 3 above)
Comment (ebraminio) [reply src/hb-ot-stat-table.hh]: Thanks. Done

File: src/hb-ot-stat-table.hh (hunk 4 above)
Comment (ebraminio) [reply src/hb-ot-stat-table.hh]: Removing whole logic for now.

**Pull request merged**